src/Catalyst.Core.Modules.Ledger/Web3EthApi.cs

```diff
@@ -22,25 +22,38 @@
 #endregion
 
 using System;
+using Autofac.Features.AttributeFilters;
+using Catalyst.Abstractions.Consensus.Deltas;
 using Catalyst.Abstractions.Kvm;
 using Catalyst.Abstractions.Ledger;
+using Nethermind.Evm;
 using Nethermind.Store;
 
 namespace Catalyst.Core.Modules.Ledger 
 {
     public class Web3EthApi : IWeb3EthApi
-    {
-        public Web3EthApi(IDeltaExecutor deltaExecutor, IStateReader stateReader, IStateRootResolver stateRootResolver, IDeltaResolver deltaResolver)
+    { 
+        public const string ComponentName = nameof(Web3EthApi);
+            
+        public Web3EthApi(IStateReader stateReader, IStateRootResolver stateRootResolver, IDeltaResolver deltaResolver, IDeltaCache deltaCache, [KeyFilter(ComponentName)] ITransactionProcessor processor, [KeyFilter(ComponentName)] IStorageProvider storageProvider, [KeyFilter(ComponentName)] IStateProvider stateProvider)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Split this 321 characters long line (which is greater than 200 authorized).](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4585091)